test/Microsoft.Extensions.Caching.Memory.Tests/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -410,37 +410,49 @@ public void SetGetAndRemoveWorksWithObjectKeysWhenDifferentReferences()
         }
 
         [Fact]
-        public void GetAndSet_AreThreadSafe()
+        public void GetAndSet_AreThreadSafe_AndUpdatesNeverLeavesNullValues()
```

Comment:
Updating this test to handle the null scenario. Could split it into a separate test but I don't think that's necessary.